knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -112,16 +123,45 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	}
 
 	gvkToResource := map[schema.GroupVersionKind]client.Object{
-		consolev1.GroupVersion.WithKind("ConsoleCLIDownload"): &consolev1.ConsoleCLIDownload{},
-		routev1.GroupVersion.WithKind("Route"):                &routev1.Route{},
+		routev1.GroupVersion.WithKind("Route"): &routev1.Route{},
+	}
+
+	// If console is installed add the ccd resource watcher, otherwise remove it avoid manager exiting due to kind not found.
+	if common.ConsoleInstalled.Load() {
+		gvkToResource[consolev1.GroupVersion.WithKind("ConsoleCLIDownload")] = &consolev1.ConsoleCLIDownload{}
+	} else {
+		enqueueRequests := handler.MapFunc(func(obj client.Object) []reconcile.Request {
+			if obj.GetName() == consoleclidownload.CLIDownloadCRDName {
+				log.Info("Serving, processing crd request", "name", obj.GetName())
+				common.ConsoleInstalled.Store(true)
+				_ = health.InstallHealthDashboard(r.(*ReconcileKnativeServing).client)
```

Comment:
We don't need this if we dont check for dashboards.

Reply:
How about Quickstarts? I think they are a CRD https://docs.openshift.com/container-platform/4.11/rest_api/console_apis/consolequickstart-console-openshift-io-v1.html

Reply:
Yes there is one I saw that when reconciled the rest but they are both installed with the console so no need to discover them explicitly.

Reply:
The CRDs are deployed at random by Console operator. Even if they are deployed sequentially, we shouldn't rely on that behavior. Now you are waiting for `ConsoleCLIDownload` to be available. There is a chance the `ConsoleQuickstart` would not be deployed yet, although a small one.

Instead of waiting for `ConsoleCLIDownload` to be available, we could wait for [`ClusterOperator`](https://docs.openshift.com/container-platform/4.11/rest_api/config_apis/clusteroperator-config-openshift-io-v1.html) named `console` to be available and reported `Progressing==False && Degraded==False`. That would make sure the console is ready.

Reply:
> Instead of waiting for ConsoleCLIDownload to be available, we could wait for [ClusterOperator](https://docs.openshift.com/container-platform/4.11/rest_api/config_apis/clusteroperator-config-openshift-io-v1.html) named console to be available and reported Progressing==False && Degraded==False.

I am not even sure that this makes sure that crds are there. Does it explicitly check again for crds status after they are created? Anyway I prefer the crd check instead.

Reply:
I checked clusteroperator too so it does get to `Progressing==False && Degraded==False` after crds are installed but will try the crds approach first as closer to what is being installed.

Reply:
Okay. Let's go with CRDs way. However, I think it may simplify the code if we instead rely on `ClusterOperator` status.